packages/@apphosting/adapter-angular/src/utils.ts

```diff
@@ -32,21 +40,113 @@ export async function loadConfig(cwd: string) {
   const target = "build";
   if (!workspaceProject.targets.has(target)) throw new Error("Could not find build target.");
 
-  const { builder, defaultConfiguration: configuration = "production" } =
-    workspaceProject.targets.get(target)!;
+  const { builder } = workspaceProject.targets.get(target)!;
   if (builder !== "@angular-devkit/build-angular:application") {
     throw new Error("Only the Angular application builder is supported.");
   }
+}
 
-  const buildTarget = {
-    project,
-    target,
-    configuration,
+// Populate file or directory paths we need for generating output directory
+export function populateOutputBundleOptions(outputPaths: OutputPaths): OutputPathOptions {
+  const outputBundleDir = resolve(".apphosting");
+
+  const baseDirectory = fileURLToPath(outputPaths["root"]);
+  const browserRelativePath = relative(baseDirectory, fileURLToPath(outputPaths["browser"]));
+  let serverRelativePath = "server";
+  if (outputPaths["server"]) {
+    serverRelativePath = relative(baseDirectory, fileURLToPath(outputPaths["server"]));
+  }
+
+  return {
+    bundleYamlPath: resolve(outputBundleDir, "bundle.yaml"),
+    outputDirectory: outputBundleDir,
+    baseDirectory,
+    outputBaseDirectory: resolve(outputBundleDir, "dist"),
+    serverFilePath: resolve(outputBundleDir, "dist", serverRelativePath, "server.mjs"),
+    browserDirectory: resolve(outputBundleDir, "dist", browserRelativePath),
   };
+}
+
+// Run build command
+export const build = (cwd = process.cwd()) =>
+  new Promise<OutputPathOptions>((resolve, reject) => {
+    // enable JSON build logs for application builder
+    process.env.NG_BUILD_LOGS_JSON = "1";
+    const childProcess = spawn("npm", ["run", "build"], {
+      cwd,
+      shell: true,
+      stdio: ["inherit", "pipe", "pipe"],
+    });
+    let outputPathOptions = {} as OutputPathOptions;
```

Comment:
instead, let's make this `let outputPathOptions: OutputPathOptions | undefined;` and on exit, throw is outputPathOptions is still undefined